contracts/CommunityVault.sol

```diff
@@ -14,15 +16,91 @@ import { CommunityOwnable } from "./CommunityOwnable.sol";
  *      Only community owners, as defined in the CommunityOwnable contract, have
  *      permissions to transfer these tokens out of the vault.
  */
-contract CommunityVault is CommunityOwnable {
+contract CommunityVault is CommunityOwnable, IERC721Receiver {
     using SafeERC20 for IERC20;
+    using EnumerableSet for EnumerableSet.UintSet;
+
+    event ERC20Deposited(address indexed depositor, address indexed token, uint256 amount);
+    event ERC721Deposited(address indexed depositor, address indexed token, uint256 tokenId);
 
     error CommunityVault_LengthMismatch();
     error CommunityVault_NoRecipients();
     error CommunityVault_TransferAmountZero();
+    error CommunityVault_ERC20TransferAmountTooBig();
+    error CommunityVault_DepositAmountZero();
+    error CommunityVault_IndexOutOfBounds();
+    error CommunityVault_ERC721TokenAlreadyDeposited();
+    error CommunityVault_ERC721TokenNotDeposited();
+
+    mapping(address => uint256) public erc20TokenBalances;
+    mapping(address => EnumerableSet.UintSet) private erc721TokenIds;
```

Comment:
```suggestion
    mapping(address owners => EnumerableSet.UintSet balances) private erc721TokenIds;
```

Just cosmetics, but I think we can use named mapping annotation here

Reply:
we need to upgrade to a newer version of the compiler to do it, shall we do it in another PR? 
in this case instead of `owners` should be `tokens`

Reply:
Yes that can be done in a different PR. Good point.